src/user_input.rs

```diff
@@ -47,6 +56,38 @@ impl UserInput {
         }
     }
 
+    /// Creates a [`UserInput::Chord`] from an iterator of [`InputKind`]s
+    ///
+    /// If `inputs` has a length of 1, a [`UserInput::Single`] variant will be returned instead.
+    ///
+    /// If `inputs` contains more than `CHORD_MAX_SIZE` elements, a [`ChordCreationError`] is returned.
+    /// If you want to silently ignore the rest of the elements, use the [`UserInput::chord`] function instead.
+    pub fn try_make_chord(
+        inputs: impl IntoIterator<Item = impl Into<InputKind>>,
+    ) -> Result<Self, ChordCreationError> {
+        // We can't just check the length unless we add an ExactSizeIterator bound :(
+        let mut length: usize = 0;
+
+        let mut set: PetitSet<InputKind, CHORD_MAX_SIZE> = PetitSet::default();
+
+        // If the iterator contains more than the maximum number of elements, ignore the rest
+        for button in inputs.into_iter().take(CHORD_MAX_SIZE) {
+            length += 1;
+
+            if length >= CHORD_MAX_SIZE {
+                return Err(ChordCreationError);
+            }
+            set.insert(button.into());
+        }
+
+        let user_input = match length {
```

Comment:
This matching behavior is inconsistent with the behavior in the infallible version. It's also surprising, as the user has explicitly tried to make a chord.

Might still be correct, but if so we'll want a different name.

Reply:
Oh, I see. Never mind!